### ISSUE TITLE
Question about video model finetuning

### ISSUE THREAD

**Issue by h6kplus** [X-GenGroup/Flow-Factory#89]
Nice work! I really appreciate your effort on creating a unified framework. 

I'm currently tring to apply these algorithms with my own reward on Wan2.1-1.3B but I'm finding difficulty even for training with rewards like pickscore. Maybe can you share some successful configs for training? Thank you so much! 

And also, is there any plan for implementing some video based scores like video align? I can also try implement it and send pull requests

Again thank you for sharing your work!

**Comment by luoxyhappy** [comment]
@Jayce-Ping hi, thanks for your great work! Did you try DiffusionNFT with wan2.1 1.3B? my reward quickly dropped at a certain step.

<img width="844" height="742" alt="Image" src="https://github.com/user-attachments/assets/01e14e5e-ffbe-4c27-a081-fdc9951b3ab0" />

**Comment by Jayce-Ping** [comment]
What is your detailed setting?

**Comment by luoxyhappy** [comment]
# Model Configuration
model:
  finetune_type: 'lora' # Options: full, lora
  lora_rank: 128
  lora_alpha: 128
  target_components: 'transformer'
  target_modules: "default"
  model_name_or_path: "Wan2.1-T2V-1.3B-Diffusers"
  model_type: "wan2_t2v"  # wan2_t2v, wan2_i2v, wan2_v2v
  resume_path: null # Path to load previous checkpoint/lora adapter
  resume_type: null # Options: lora, full, state. Null to auto-detect based on `finetune_type`
  # attn_backend: '_flash_3_hub' # Use flash attention 3 backend.

log:
  run_name: null  # Run name (auto: {model_type}_{finetune_type}_{trainer_type}_{timestamp})
  project: "Flow-Factory"  # Project name for logging
  logging_backend: "tensorboard"  # Options: wandb, swanlab, tensorboard, none
  save_dir: "/mnt/workspace/luoxiangyang/Flow-Factory/saves/"
  save_freq: 20  # Save frequency in epochs (0 to disable)
  save_model_only: true  # Save only the model weights (not optimizer, scheduler, etc.)

# Training Configuration
train:
  # Trainer settings
  trainer_type: 'nft'
  advantage_aggregation: 'gdpo' # Options: 'sum', 'gdpo'
  nft_beta: 0.1
  # `Old` Policy settings
  off_policy: true # Whether to use ema parameters for sampling off-policy data.
  ema_decay_schedule: "piecewise_linear"  # Decay schedule for EMA. Options: ['constant', 'power', 'linear', 'piecewise_linear', 'cosine', 'warmup_cosine']
  flat_steps: 0
  ramp_rate: 0.001
  ema_decay: 0.5  # EMA decay rate (0 to disable)
  ema_update_interval: 1  # EMA update interval (in epochs)
  ema_device: "cuda"  # Device to store EMA model (options: cpu, cuda)
  # Training Timestep distribution
  num_train_timesteps: 4 # Set null to all steps
  time_sampling_strategy: discrete # Options: uniform, logit_normal, discrete, discrete_with_init, discrete_wo_init
  time_shift: 3.0
  timestep_range: 0.9
  # KL div
  kl_type: 'v-based'
  kl_beta: 0 # KL divergence beta, 0 to disable
  ref_param_device: 'cuda' # Options: cpu, cuda
  # Clipping
  adv_clip_range: 5.0  # Advantage clipping range

  # Sampling Settings
  resolution: [480, 832]  # Can be int or [height, width]
  condition_image_size: [480, 832]  # Condition image size for sampling
  num_frames: 1 # Training frames
  num_inference_steps: 12  # Number of timesteps
  guidance_scale: 5.0  # Guidance scale for sampling

  # Batch and sampling
  per_device_batch_size: 1  # Batch size per device
  group_size: 16  # Group size for GRPO sampling
  global_std: false  # Use global std for advantage normalization
  unique_sample_num_per_epoch: 48  # Unique samples per group
  gradient_step_per_epoch: 1  # Gradient steps per epoch
  gradient_accumulation_steps: auto  # Options: auto, or positive integer. When set, `gradient_step_per_epoch` is ignored.
  
  # Optimization
  seed: 42  # Random seed
  learning_rate: 1.0e-4  # Initial learning rate
  adam_weight_decay: 1.0e-4  # AdamW weight decay
  adam_betas: [0.9, 0.999]  # AdamW betas
  adam_epsilon: 1.0e-8  # AdamW epsilon
  max_grad_norm: 1.0  # Max gradient norm for clipping

  # Gradient checkpointing
  enable_gradient_checkpointing: false  # Enable gradient checkpointing to save memory with extra compute


# Scheduler Configuration
scheduler:
  dynamics_type: "ODE"  # Options: Flow-SDE, Dance-SDE, CPS, ODE

# Evaluation settings
eval:
  resolution: [480, 832]  # Evaluation resolution
  num_frames: 41 # Evaluation frames
  per_device_batch_size: 1  # Eval batch size
  guidance_scale: 5.0  # Guidance scale for sampling
  num_inference_steps: 20  # Number of eval timesteps
  eval_freq: 20  # Eval frequency in epochs (0 to disable)
  seed: 42  # Eval seed (defaults to training seed)

# Reward Model Configuration
rewards:
  - name: "pick_score"
    reward_model: "PickScore"
    batch_size: 16
    device: "cuda"
    dtype: bfloat16

# Optional Evaluation Reward Models
eval_rewards:
  - name: "pick_score"
    reward_model: "PickScore"
    batch_size: 32
    device: "cuda"
    dtype: bfloat16

**Comment by Jayce-Ping** [comment]
配置看着挺正常的。请问采样出来的视频正常吗？如果视频不正常的话可能学不到东西。此外，还可以试试把`nft_beta`设置为1试试.
